File: DAA/Assignment_1/search_sum.c
#include <stdio.h>
int main()
{
    int n;
    printf("Enter number of elements:");
    scanf("%d", &n);

    int X;
    printf("Enter search sum number:");
    scanf("%d", &X);

    int arr[n];
    printf("Enter elements in sorted order|n");
    for(int i=0; i<n; i++)
    {
        printf("Enter element:");
        scanf("%d", &arr[i]);
    }

    int left = 0, right = n-1, f = 0;
    while(left < right)
    {
        int sum = arr[left] + arr[right];
        if(sum == X)
        {
            f = 1;
            break;
        }
        else if(sum < X)
        left++;
        else
        right--;
    }

    if(f == 1)
    printf("%d and %d sums up to %d", arr[left], arr[right], X);
    else
    printf("Sum not found!");

    return 0;
}

File: DAA/Assignment_1/search_sum.py
n = int(input("Enter number of elements:"))
X = int(input("Enter search sum number:"))
arr = []
print("Enter elements in sorted order")
for i in range(n):
    arr.append(int(input("Enter element:")))

left = 0
right = n-1
f = 0
while left < right:
    sum = arr[left] + arr[right]
    if sum == X:
        f = 1
        break
    elif sum < X:
        left += 1
    else:
        right -= 1

if f == 1:
    print(f"{arr[left]} and {arr[right]} sums up to {X}")
else:
    print("Sum not found!")